where-to-travel/src/main/webapp/signup.js

```diff
@@ -0,0 +1,61 @@
+// Get elements for authentication
+const emailSignUp = document.getElementById('emailSignUp');
+const passwordSignUp = document.getElementById('passwordSignUp');
+const passwordConfirmation = document.getElementById('passwordConfirmation');
+const btnSignUp = document.getElementById('signUp');
+
+// Add signup event
+btnSignUp.addEventListener('click', e => {
+ 
+  const email = emailSignUp.value;
+  const pass = passwordSignUp.value;
+  const passConfirmation = passwordConfirmation.value;
+  const auth = firebase.auth();
+
+  if (validate()==false) {
+    return;
+  } 
+  else if(pass != passConfirmation) {
+    alert("Your passwords do not match. Please try again.");
+  }
+  // Password must be at least 6 characters in length
+  else if(pass.length < 6) {
+    alert("Your password must be at least 6 characters long. Please try again");
+  }
+  else {
+    // Sign up
+    const promise = auth.createUserWithEmailAndPassword(email, pass);
+    promise.then(e => {
+      alert("You have sucessfully signed up!");
+      
+      // Add user information to the real time database in Firebase
+      let database = firebase.database();
+      let ref = database.ref('users');
+        let data = {
+          email: email,    
+          uID: auth.currentUser.uid
+        }
+     ref.push(data);
+    });
+    promise.catch(e => console.log(e.message), alert(e.message));
+  }
+});
+
+// Verify that the email address is properly formatted using a regular expression.
+// A valid email is a string (a subset of ASCII characters) separated into two parts by an @ symbol. 
+// The two parts being personal_info and a domain, that is personal_info@domain. 
+function validateEmail(email) {
+  const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
This line is long, can we split it with a multi line string?

Reply:
good idea. done